tpke/src/ciphertext.rs

```diff
@@ -93,43 +93,52 @@ pub fn encrypt<R: RngCore, E: PairingEngine>(
         .mul(rand_element)
         .into();
 
+    // TODO: Consider adding aad to the Ciphertext struct
     Ciphertext::<E> {
         commitment,
         ciphertext,
         auth_tag,
     }
 }
 
+/// Implements the check section 4.4.2 of the Ferveo paper, 'TPKE.CheckCiphertextValidity(U,W,aad)'
+/// See https://eprint.iacr.org/2022/898.pdf
 pub fn check_ciphertext_validity<E: PairingEngine>(
     c: &Ciphertext<E>,
     aad: &[u8],
-) -> bool {
+) -> Result<()> {
     let g_inv = E::G1Prepared::from(-E::G1Affine::prime_subgroup_generator());
```

Comment:
Can we take this from `SetupParams`?

Reply:
Yes, we just need to pass `g_inv` through a lot of places. See my latest commit for details.